packages/utils/src/Interfaces.ts

```diff
@@ -63,7 +63,7 @@ export interface ExecutionRequest<
   TExtensions = Record<string, any>
 > {
   document: DocumentNode;
-  operationType: OperationTypeNode;
+  operationType?: OperationTypeNode;
```

Comment:
Instead of making this optional, we can document the new util to get the operationType and pass it because otherwise all executor implementation will need to have to check if operationType is given and visit the document to get it everytime it is needed.

Reply:
Sorry, I don't follow the specifics here... You're saying to keep this required and mandate devs use a new util to fill it in? Personally, my motivations for making it optional was so that devs didn't need to do that, thus making the tool simpler to use. Concerning performance, we have two scenarios:

1. Internal GraphQL Tools operations (stitching, etc). These automatically populate this field anyway, so shouldn't have any regressions in hot code paths.
2. Manual developer calls... these provide the convenience of filling in the information automatically. The developer can provide the value if they're overly concerned with performance.

Alternatively, we could memoize the resolved operationType on each request, but @yaacovCR had questioned the merits of that earlier in this PR.

Reply:
Okay, good feedback. I've refactored it so that [operationType is always assessed up front](https://github.com/ardatan/graphql-tools/pull/3588/files#diff-d4824895ea613dcc1f710c3ac82e952fe0ca12391b671f70d9f2d90d5656fdceR25-R29), so there's a maximum of one document check per request. Then, [mergeRequests receives an operation type](https://github.com/ardatan/graphql-tools/pull/3588/files#diff-e6729f0d54dbdfb10a93bf6592cf82c8ea91222f9de3aaf271a45974df6fd08cR60) so that there's no guesswork about where to look to decide what operation is being built. Does this address your concerns?

Reply:
Actually `operationType` is concern of the developer rather than stitching internals because for example in that case I'd need to update the following parts because we should ensure operationType is given if not get it from operation.
https://github.com/ardatan/graphql-tools/blob/master/packages/delegate/src/delegateToSchema.ts#L227
https://github.com/ardatan/graphql-tools/blob/master/packages/loaders/url/src/index.ts#L667

Reply:
@ardatan okay, I've restored `operationType` as a required field. That said, I did leave `operationType` as a functional argument going into the `mergeRequests` method so that it no longer gives operational priority to any one request. All operation type concerns are now localized to the `batchExecuteLoadFn` method, which is future facing to a time when `operationType` is hopefully removed entirely.

Have another look!